src/property.ts

```diff
@@ -32,6 +36,8 @@ export function hasNestedProperty(prop: Property) {
     case Property.TYPE:
     case Property.BIN_MAXBINS:
     case Property.SCALE_TYPE:
+    case Property.TRANSFORM_CALCULATE:
```

Comment:
I don't know why you have to add TRANSFORM_CALCULATE now.  

`TRANSFORM_CALCULATE` clearly have nested property  (`field` and `expr`) ?